src/NSL/IO/h5_io.tpp

```diff
@@ -406,18 +406,30 @@ class H5IO {
         inline bool overwrite(){
             return overwrite_;
         } // overwrite()
-          
-    private:
+        
         //! Removes a group if overwrite == True and group exists
-        void removeData_(std::string node){
+        void removeData(std::string node){
             bool exist = this->exist(node);
             // remove the group if it exists; once the file is closed
             // automatic repacking is applied
             if (overwrite_ and exist){
                 NSL::Logger::debug("Unlinking Dataset (overwrite={}; node exists={}): {}",overwrite_,exist,node);
                 h5f_.unlink(node);
             }
+        } 
+
+        //! Removes a group if group exists
+        void trimData(std::string node){
```

Comment:
Should we maybe merge `removeData` and `trimData`. The only difference is the overwrite_ flag. If this is an undesired/faulty feature should remove it

Reply:
I made a new function called deleteData which is basically trimData but I wanted the name to be more descriptive.

delteData deletes a given node if it exists in the file.